src/Http/Http.Features/src/FeatureReferences.cs

```diff
@@ -108,5 +108,17 @@ public void Initalize(IFeatureCollection collection, int revision)
 
         public TFeature Fetch<TFeature>(ref TFeature cached, Func<IFeatureCollection, TFeature> factory)
             where TFeature : class => Fetch(ref cached, Collection, factory);
+
+        private static int ContextDisposed()
```

Comment:
Curious: Why add these extra methods instead of changing their use above to
``` c#
if (Collection?.Revision == null)
{
    throw new ObjectDisposedException("HttpContext", "Request has finished and HttpContext disposed.");
}
```

Reply:
Inlining. 

`Load string, Load string, New Exception, Throw` is quite a chunky bit of code to inline to every caller